### ISSUE TITLE
No Support for ItemReader in Map State of StateMachine definition

### ISSUE THREAD

**Issue by apalmer1377** [aws-cloudformation/cfn-lint#2849]
### CloudFormation Lint Version

0.79.7

### What operating system are you using?

Mac

### Describe the bug

I receive the following error when trying to include `ItemReader` in a `Map` state:

```
E2532 State Machine Definition key (ItemReader) for State (Map) of Type (Map) is not valid
service.template:8:7
```

### Expected behavior

However, `ItemReader` is a valid field for a Map State, according to the AWS States Language: https://states-language.net/
Including an `ItemReader` field should not throw an error.

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"

Resources:
  WareHouseLoadStateMachine:
    Type: AWS::StepFunctions::StateMachine
    Properties:
      RoleArn: role-arn
      DefinitionString: |
          {
            "Comment": "map",
            "TimeoutSeconds": 72000,
            "StartAt": "Map",
            "States":
            {
              "Map": {
                "Type": "Map",
                "ItemProcessor": {},
                "ItemReader": {
                  "Resource": "arn:aws:states:::s3:getObject",
                  "ReaderConfig": {
                    "InputType": "JSON"
                  },
                  "Parameters": {
                    "Bucket": "mybucket",
                    "Key": "mykey"
                  }
                },
                "MaxConcurrency": 1000,
                "End": true              
              }
            }
          }
```

**Comment by kddejong** [comment]
Released.
